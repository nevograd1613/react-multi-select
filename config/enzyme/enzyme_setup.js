// TODO: Remove this `raf` polyfill once the below issue is sorted
// https://github.com/facebookincubator/create-react-app/issues/3199#issuecomment-332842582
import raf from "./tempPolyfills";

import Enzyme from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";

Enzyme.configure({ adapter: new Adapter() });
