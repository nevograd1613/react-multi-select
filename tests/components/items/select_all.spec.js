import React from "react";
import ShallowRenderer from "react-shallow-renderer";
import { shallow } from "enzyme";

import SelectAll from "../../../src/components/items/select_all";

//   selectedIds

describe("SelectAll", () => {
  test("default snapshot", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<SelectAll />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with selectAllMessage", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <SelectAll selectAllMessage="some test message" />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with isSelectAll", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<SelectAll isSelectAll={true} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with isSelectAll false", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<SelectAll isSelectAll={false} />);
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with empty selectedIds with select all", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <SelectAll selectedIds={[]} isSelectAll={true} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with selectedIds with select all should be indeterminate", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <SelectAll selectedIds={[1]} isSelectAll={true} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with empty selectedIds without select all", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <SelectAll selectedIds={[]} isSelectAll={false} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with selectedIds without select all should be indeterminate", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
      <SelectAll selectedIds={[1]} isSelectAll={false} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test("snapshot with height", () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<SelectAll height={20} />);
    expect(tree).toMatchSnapshot();
  });

  test("click will trigger onClick", () => {
    const onClick = jest.fn();
    const item = shallow(<SelectAll onClick={onClick} />);
    item.simulate("click");
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
