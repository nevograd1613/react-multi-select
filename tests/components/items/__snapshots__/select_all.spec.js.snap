// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectAll default snapshot 1`] = `
<Item
  checked={false}
  disabled={false}
  height={40}
  indeterminate={false}
  item={
    {
      "label": "Select All",
    }
  }
  onClick={undefined}
  withBorder={true}
/>
`;

exports[`SelectAll snapshot with empty selectedIds with select all 1`] = `
<Item
  checked={false}
  disabled={false}
  height={40}
  indeterminate={false}
  item={
    {
      "label": "Select All",
    }
  }
  onClick={undefined}
  withBorder={true}
/>
`;

exports[`SelectAll snapshot with empty selectedIds without select all 1`] = `
<Item
  checked={false}
  disabled={false}
  height={40}
  indeterminate={false}
  item={
    {
      "label": "Select All",
    }
  }
  onClick={undefined}
  withBorder={true}
/>
`;

exports[`SelectAll snapshot with height 1`] = `
<Item
  checked={false}
  disabled={false}
  height={20}
  indeterminate={false}
  item={
    {
      "label": "Select All",
    }
  }
  onClick={undefined}
  withBorder={true}
/>
`;

exports[`SelectAll snapshot with isSelectAll 1`] = `
<Item
  checked={false}
  disabled={false}
  height={40}
  indeterminate={false}
  item={
    {
      "label": "Select All",
    }
  }
  onClick={undefined}
  withBorder={true}
/>
`;

exports[`SelectAll snapshot with isSelectAll false 1`] = `
<Item
  checked={false}
  disabled={false}
  height={40}
  indeterminate={false}
  item={
    {
      "label": "Select All",
    }
  }
  onClick={undefined}
  withBorder={true}
/>
`;

exports[`SelectAll snapshot with selectAllMessage 1`] = `
<Item
  checked={false}
  disabled={false}
  height={40}
  indeterminate={false}
  item={
    {
      "label": "some test message",
    }
  }
  onClick={undefined}
  withBorder={true}
/>
`;

exports[`SelectAll snapshot with selectedIds with select all should be indeterminate 1`] = `
<Item
  checked={false}
  disabled={false}
  height={40}
  indeterminate={true}
  item={
    {
      "label": "Select All",
    }
  }
  onClick={undefined}
  withBorder={true}
/>
`;

exports[`SelectAll snapshot with selectedIds without select all should be indeterminate 1`] = `
<Item
  checked={false}
  disabled={false}
  height={40}
  indeterminate={true}
  item={
    {
      "label": "Select All",
    }
  }
  onClick={undefined}
  withBorder={true}
/>
`;
